src/lsm/groove.zig

```diff
@@ -1119,13 +1137,14 @@ pub fn GrooveType(
 
             inline for (std.meta.fields(IndexTrees)) |field| {
                 const Helper = IndexTreeFieldHelperType(field.name);
-
-                const index = Helper.derive_index(object);
-                if (index != 0 or comptime index_zeroes(field)) {
-                    @field(groove.indexes, field.name).put(&.{
-                        .timestamp = object.timestamp,
-                        .field = index,
-                    });
+                if (Helper.derive_index(object)) |value| {
+                    const allow_zero = comptime index_zeroes(field);
+                    if (allow_zero or value != 0) {
+                        @field(groove.indexes, field.name).put(&.{
+                            .timestamp = object.timestamp,
+                            .field = value,
+                        });
+                    }
```

Comment:
:thinking: It is a little weird that derived indexes now have 2 different ways to be optional.
Maybe we should _require_ that derived indexes allow zero values?

(I recognize that for `expires_at` and `imported` it doesn't make a difference one way or the other, since they are nonzero.)

Reply:
Since `derive_index` is the common abstraction for dealing with fields and functions, we could unify the way to represent optional indexes always as `null`.

What do you think if the function `derive_index` (maybe renamed as `index_from_object`) returned `null` instead of zero for optional fields (and the "optional" list would be only applicable to fields, not to derived indexes)?